lib/Runtime/Library/WebAssemblyMemory.cpp

```diff
@@ -139,9 +139,34 @@ WebAssemblyMemory::GrowInternal(uint32 deltaPages)
         return -1;
     }
 
-    ArrayBuffer * newBuffer = m_buffer->TransferInternal(newBytes);
-    m_buffer = newBuffer;
+    ArrayBuffer * newBuffer = nullptr;
+    JavascriptExceptionObject* caughtExceptionObject = nullptr;
+    try
+    {
+        newBuffer = m_buffer->TransferInternal(newBytes);
+    }
+    catch (const JavascriptException& err)
+    {
+        caughtExceptionObject = err.GetAndClear();
+        Assert(caughtExceptionObject);
+        //Propagate if not OOM
+        if (caughtExceptionObject != ThreadContext::GetContextForCurrentThread()->GetPendingOOMErrorObject())
+        {
+            caughtExceptionObject = caughtExceptionObject->CloneIfStaticExceptionObject(GetScriptContext());
+            JavascriptExceptionOperators::DoThrow(caughtExceptionObject, GetScriptContext());
+        }
+        GetRecycler()->ReportExternalMemoryAllocation(oldBytes); //transferInternal overestimates the amount of freed memory
```

Comment:
I don't understand the case where this can happen.
If this really happens then it's a bug in TransferInternal, not in GrowMemory